programs/ssl/ssl_client2.c

```diff
@@ -535,6 +545,81 @@ static int eap_tls_key_derivation ( void *p_expkey,
     }
     return( 0 );
 }
+
+static int nss_keylog_export( void *p_expkey,
+                              const unsigned char *ms,
+                              const unsigned char *kb,
+                              size_t maclen,
+                              size_t keylen,
+                              size_t ivlen,
+                              unsigned char client_random[32],
+                              unsigned char server_random[32],
+                              mbedtls_tls_prf_types tls_prf_type )
+{
+    char nss_keylog_line[ 200 ];
+    size_t const client_random_len = 32;
+    size_t const master_secret_len = 48;
+    size_t len = 0;
+    size_t j;
+    int ret = 0;
+
+    ((void) p_expkey);
+    ((void) kb);
+    ((void) maclen);
+    ((void) keylen);
+    ((void) ivlen);
+    ((void) server_random);
+    ((void) tls_prf_type);
+
+    len += sprintf( nss_keylog_line + len,
```

Comment:
>  If you prefer keeping it for the sake of consistency, I don't mind it. However, this may be misleading to whoever sees the code for the first time.
Yes, it's for the sake of consistency.

Reply:
> What are the chances of this of any of the following sprintf in this function overflowing nss_keylog_line?

The length of the string is statically determined: We have `CLIENT RANDOM `, and then the hex presentations of the 48 + 32 = 80 Bytes from Master Secret and the Random Bytes. We're encoding them via two characters per byte, so need 160 characters for that. So the 200 characters reserved in `nss_keylog_line` are enough as far as I can see.